frontend/packages/console-plugin-shared/src/components/Markdown/MarkdownView.tsx

```diff
@@ -183,39 +184,30 @@ const IFrameMarkdownView: React.FC<InnerSyncMarkdownProps> = ({
   theme,
   updateThemeClass,
 }) => {
-  const [frame, setFrame] = React.useState<HTMLIFrameElement>();
+  const frameRef = React.useRef<HTMLIFrameElement>(null);
   const [frameHeight, setFrameHeight] = React.useState(0);
   const [loaded, setLoaded] = React.useState(false);
-  const htmlTagElement = frame?.contentDocument?.documentElement;
 
   // eslint-disable-next-line react-hooks/exhaustive-deps
   const updateDimensions = React.useCallback(
     _.debounce(() => {
-      if (!frame?.contentWindow?.document?.body?.firstElementChild) {
+      if (!frameRef?.current?.contentWindow) {
```

Comment:
Here I might have to keep the second optional for the `current` as follows`!frameRef.current?.contentWindow`
One of the integration test was failing after making this change due to `reading contentWindow from undefined` error. 
I think this occurs when the iframe is not fully mounted yet ( frameRef.current is still null), so the `useResizeObserver` fallbacks to a `<body>` element instead of the iframe and the callback to `updateDimensions` tries to read `contentWindow`, which is undefined at that point.

Reply:
Agreed, in that case you could also change the code from
```ts
if (!frameRef.current?.contentWindow) {
  return;
}
setFrameHeight(
  frameRef.current.contentWindow.document.body.firstElementChild.scrollHeight + (exactHeight ? 0 : 15),
);
```
to
```ts
if (frameRef.current?.contentWindow) {
  setFrameHeight(
    frameRef.current.contentWindow.document.body.firstElementChild.scrollHeight + (exactHeight ? 0 : 15),
  );
}
```